src/basic/compress.h

```diff
@@ -18,15 +39,15 @@ int compress_blob_zstd(const void *src, uint64_t src_size,
 static inline int compress_blob(const void *src, uint64_t src_size,
                                 void *dst, size_t dst_alloc_size, size_t *dst_size) {
         int r;
-#if DEFAULT_COMPRESSION == COMPRESSION_ZSTD
+#if defined(DEFAULT_COMPRESSION_ZSTD)
```

Comment:
conf10 doesn't work here, because the macro name is generated dynamically, i.e. if zstd is the default, then DEFAULT_COMPRESSION_ZSTD is set, but DEFAULT_COMPRESSION_LZ4 is not set whatsoever...

Reply:
I prepared https://github.com/systemd/systemd/pull/23160 to try to simplify the definitions. I think it should be merged first and this PR rebased.